### PR TITLE
OOP school library: add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by isaacpitwa**
## OOP school library: add Person, Student and Teacher classes
- Create class `Person` with  the following:
  - Instance vars: `@id`, `@name`, and `@age`.
  - Constructor with `name`,  `age`, and `parent_permission` as parameter. `name` and `parent_permission` are optional and have default values of `"Unknown"` and `true`.
  - Getters for `@id`, `@name`, and `@age`.
  - Setters for `@name` and `@age`.
  - Private method `is_of_age?` that returns `true` if `@age` is greater or equal to 18 and `false` otherwise.
  - Public method `can_use_services?` that returns `true` if person is of age or if they have permission from parents.
- Create class `Student` with the following:
  - Inherits from `Person`.
  - Constructor extends parent's constructor by adding `@classroom` and a parameter for it.
  - Method `play_hooky` that returns `"¯\(ツ)/¯"`.
- Create class `Teacher` with the following:
  - Inherits from `Person`.
  - Constructor extends parent's constructor by adding `@specialization` and a parameter for it.
  - Override `can_use_services?` so it always returns `true`.
- Each class should be saved in a separate file.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  attr_reader :id
+
+  attr_accessor :name, :age
+
+  def of_age
+    @age >= 18
+  end
+  private :of_age
+
+  def can_use_services
```
Comment (rustam-tolipov) [inline person.rb:13]: - [x] Please move `can_use_services` before than private method.

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  end
+
+  # Setters
+
+  # Getters
+  attr_reader :id
+
+  attr_accessor :name, :age
+
+  def of_age
```
Comment (rustam-tolipov) [inline person.rb:9]: - [x] Please put all private methods after the `private` declaration.

File: person.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    @age >= 18
+  end
+  private :of_age
+
+  def can_use_services
+    of_age | @parent_permission
```
Comment (rustam-tolipov) [inline person.rb]: - [x] This boolean operation is used rareil. I would suggest you change `|` to common case `||` to make your code better.

Explanation:

> | - (Condition 1 | Condition 2): checks both cases even if case 1 is true
> 
> || - (Condition 1 || Condition 2): doesn't bother to check the second case if the first one is true
> 
> || - operator will make your code run faster, professionally | is rarely used

File: teacher.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+require './person'
+class Teacher < Person
+  def initialize(name, parent_permission, age, specialization)
```
Comment (rustam-tolipov) [inline teacher.rb]: - [ ] Please give `name` a default value!
`name = "Unknown"`

File: student.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+require './person'
+class Student < Person
+  def initialize(name, parent_permission, age, classroom)
```
Comment (rustam-tolipov) [inline student.rb]: - [x] Please give `name` a default value!
`name = "Unknown"`

**Review by rustam-tolipov — CHANGES_REQUESTED** [review]
Hi @isaacpitwa, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
_Your project is almost complete!_
_No Linters errors_


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: teacher.rb (hunk 4 above)
Comment (isaacpitwa) [reply teacher.rb]: Thank you a lot for this guidance @Rustamxon7.
just inquiring, wouldn't the parent class emphasize this on Initialization though?

**Review by KayLemba — APPROVED** [review]
# STATUS: APPROVED :green_circle: :clap: 

Hello  @isaacpitwa        , 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

HIGHLIGHTS:

- Your project is professional 
- Your repo is professional 
- Well structured files 


Now you can merge this branch and move on :1st_place_medal: 
HAPPY CODING AND GOOD LUCK WITH THE COMMING ONES :crossed_fingers: :four_leaf_clover: 

well-1

[OPTIONAL] Kindly add a screenshot to your README file showing your webpage user Interface. You can use extentions like fireshot to capturer the entire page.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**